CorsixTH/Lua/map.lua

```diff
@@ -738,6 +738,22 @@ function Map:getParcelTileCount(parcel)
   return self.parcelTileCounts[parcel] or 0
 end
 
+--! Get the starting values of balance and reputation for this level
+--!return start (table)
+function Map:getLevelStartState()
+  local level_config, level, start = self.level_config, self.level_number, {}
+  if level_config.towns and level_config.towns[level] then
+    start["balance"] = level_config.towns[level].StartCash
+    start["reputation"] = level_config.towns[level].StartRep
+  elseif level_config.town then
+    start["balance"] = level_config.town.StartCash
+    start["reputation"] = level_config.town.StartRep
+  end
```

Comment:
```suggestion
  local towns
  if level_config.towns then
    if level_config.towns[level] then
      towns = level_config.towns[level]
    else
      towns = level_config.towns
    end
    
    start.balance = towns.StartCash or 0
    start.reputation = StartRep or 500
  end
```
Instead of copy/pasting a long sequence of indexing operations each time, introduce a variable that keeps the intermediate table available.

Also, by definition `x["abc"]` for table `x` is equivalent to `x.abc` but it has much less clutter. The rectangular brackets are useful for indexing with a variable like `[level]` above, or when your data is not an identifier, eg `x.["text with spaces"]`.